language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/AnnotationNodeContext.java

```diff
@@ -71,53 +71,112 @@ public AnnotationNodeContext() {
                 }));
-        completionItems.addAll(this.getAnnotationsInModule(context, attachedNode, pkgAliasMap));
+        completionItems.addAll(this.getCurrentModuleAnnotations(context, attachedNode, pkgAliasMap));
         return completionItems;
     }
 
-    private List<LSCompletionItem> getAnnotationsInModule(LSContext ctx, SyntaxKind kind,
-                                                          Map<String, String> pkgAliasMap) {
+    private boolean withAlias(LSContext context, AnnotationNode node, PackageID annotationOwner) {
+        PackageID currentModule = context.get(DocumentServiceKeys.CURRENT_PACKAGE_ID_KEY);
+        String orgName = annotationOwner.orgName.getValue();
+        String value = annotationOwner.getName().getValue();
+        return node.annotReference().kind() != SyntaxKind.QUALIFIED_NAME_REFERENCE
+                && !currentModule.name.value.equals(annotationOwner.name.value) && !("ballerina".equals(orgName)
+                && "lang.annotations".equals(value));
+    }
+
+    private List<LSCompletionItem> getCurrentModuleAnnotations(LSContext ctx, SyntaxKind kind,
+                                                               Map<String, String> pkgAliasMap) {
         BLangPackage bLangPackage = ctx.get(DocumentServiceKeys.CURRENT_BLANG_PACKAGE_CONTEXT_KEY);
-        List<LSCompletionItem> completionItems = new ArrayList<>();
-        List<BLangAnnotation> annotations = bLangPackage.topLevelNodes.stream()
+        List<BAnnotationSymbol> annotations = bLangPackage.topLevelNodes.stream()
                 .filter(topLevelNode -> topLevelNode instanceof BLangAnnotation)
-                .map(topLevelNode -> (BLangAnnotation) topLevelNode)
+                .map(topLevelNode -> (BAnnotationSymbol) ((BLangAnnotation) topLevelNode).symbol)
+                .collect(Collectors.toList());
+
+        return this.getAnnotationCompletionsForSymbols(ctx, annotations, kind, pkgAliasMap);
+    }
+
+    private List<LSCompletionItem> getAnnotationsInModule(LSContext context, String alias, SyntaxKind kind,
+                                                          Map<String, String> pkgAliasMap) {
+        Optional<Scope.ScopeEntry> moduleEntry = CommonUtil.packageSymbolFromAlias(context, alias);
+        if (!moduleEntry.isPresent()) {
+            List<LSCompletionItem> completionItems = new ArrayList<>();
+            // Import statement has not been added. Hence try resolving from the annotation cache
+            LSAnnotationCache.getInstance().getAnnotationsInModule(context, alias, kind)
+                    .forEach((key, value) -> value.forEach(annotation -> {
+                        completionItems.add(CommonUtil.getAnnotationCompletionItem(key, annotation, context, false,
+                                pkgAliasMap));
+                    }));
+
+            return completionItems;
+        }
+
+        List<BAnnotationSymbol> collect = moduleEntry.get().symbol.scope.entries.values().stream()
```

Comment:
Shall we rename this into a more meaningful name like `annotations` or `filteredAnnotations`